programs/benchmark/Benchmark.cpp

```diff
@@ -468,14 +475,18 @@ class Benchmark : public Poco::Util::Application
     {
         Stopwatch watch;
 
-        ConnectionPool::Entry entry = connections[connection_index]->get(
-            ConnectionTimeouts::getTCPTimeoutsWithoutFailover(settings));
+        ConnectionPool::Entry entry = connections[connection_index]->get(ConnectionTimeouts::getTCPTimeoutsWithoutFailover(settings));
+
+        bool should_reconnect = false;
+        {
+            std::lock_guard lock(queries_per_connection_mutex);
+            should_reconnect = reconnect > 0 && (++queries_per_connection[connection_index] % reconnect == 0);
```

Comment:
Also, thinking about the current implementation again. It looks like we already track `queries_executed`. Instead of tracking queries per connection, could this be as simple as `queries_executed % reconnect == 0` ?

Reply:
nvm, in that case, though simpler, I guess that the reconnections might be way less uniform across the various connections from the pool.